### PR TITLE
Module 1. Core concepts

### PR DESCRIPTION
**Pull request by boryskrutskykh**
Module 1 core concepts

### CODE REVIEW

File: src/components/index.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import Counter from "./Counter";
```
Comment (Badyanchik) [inline src/components/index.ts]: I guess you can try something like:
`export { default as Counter } from "./Counter"`
and so on

File: src/components/select/GenreSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    selectedGenre: string;
+    onSelect: (genre: string) => void;
+}
+
+const GenreSelect: React.FC<GenreSelectProps> = ({genres, selectedGenre, onSelect}) => {
+    return (
+        <div style={{paddingTop: "2%", paddingLeft: "3%", textAlign: "left"}}>
+            {genres.map((genre) => (
+                <span
+                    className={`${styles.selectButtons} ${selectedGenre === genre ? styles.selected : ''}`}
```
Comment (Badyanchik) [inline src/components/select/GenreSelect.tsx:15]: I'd recommend checking the following library:
https://www.npmjs.com/package/classnames

File: src/components/select/GenreSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                    key={genre}
+                    onClick={() => onSelect(genre)}
+                    style={{ cursor: 'pointer' }}
```
Comment (Badyanchik) [inline src/components/select/GenreSelect.tsx]: I guess you already have the 'cursor' rule in the CSS file, no?

File: src/components/select/GenreSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+export interface GenreSelectProps {
+    genres: string[];
+    selectedGenre: string;
+    onSelect: (genre: string) => void;
+}
+
+const GenreSelect: React.FC<GenreSelectProps> = ({genres, selectedGenre, onSelect}) => {
+    return (
+        <div style={{paddingTop: "2%", paddingLeft: "3%", textAlign: "left"}}>
```
Comment (Badyanchik) [inline src/components/select/GenreSelect.tsx]: Why not move those styles to the CSS file?

File: src/App.tsx [hunk 5]
```diff
@@ -1,26 +1,36 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+
+    const genres = ['All', 'Documentary', 'Comedy', 'Horror', 'Crime'];
```
Comment (Badyanchik) [inline src/App.tsx]: Well, I'd recommend a few things:
1. 'genres' is a constant, right? I understand that we are not going to change it, right?. So, I suggest moving it out of the App component (you can create a separate file with constants, or keep it in this file, but not inside the component). Why? Well, you have a state inside the App component. It means, that when the state is updated - the component will be re-rendered, so your 'genres' constant will be created again.

2. Ideally, I'd recommend creating an enum with the genres, but it's another topic, we can discuss it on a sync-up.

File: src/App.tsx [hunk 6]
```diff
@@ -1,26 +1,36 @@
...
+
+    const handleSelectGenre = (genre: string) => {
+        setSelectedGenre(genre);
+    };
+
+    return (
+        <div className="App">
+            <header>
+                <div>
+                    <p className="logo"><b style={{fontWeight: 900}}>netflix</b>roulette</p>
```
Comment (Badyanchik) [inline src/App.tsx]: I'd recommend to keep the consistency while writing CSS for the app. I mean, if you decided to use separated CSS files - write styles there, if you decided to write styles directly in JSX (ok, you can do that, but usually, we try not to do that) - do it this way. In that way, it's more easier to find an issue or change the styles.

File: src/components/select/GenreSelect.tsx (hunk 3 above)
Comment (boryskrutskykh) [reply src/components/select/GenreSelect.tsx]: Removed

File: src/components/index.ts (hunk 1 above)
Comment (boryskrutskykh) [reply src/components/index.ts]: Updated

File: src/App.tsx (hunk 5 above)
Comment (boryskrutskykh) [reply src/App.tsx]: Added updates

File: src/App.tsx (hunk 6 above)
Comment (boryskrutskykh) [reply src/App.tsx]: Updated.

**Pull request merged**